### PR TITLE
GCW 3397 Share offers on admin

### PR DESCRIPTION
**Pull request by patrixr**
ADDS:

- Menu to share offers and their packages
- Chatrooms with charities

### CODE REVIEW

File: app/controllers/review_offer.js [hunk 1]
```diff
@@ -13,6 +13,44 @@ export default Ember.Controller.extend(AsyncTasksMixin, {
...
+  }),
+
+  offerShareable: Ember.computed(
+    "offer.id",
+    "allShareables.[]",
+    "allShareables.@each.active",
+    function() {
+      return this.get("allShareables").find(sh => {
+        return (
+          sh.get("resourceType") == "Offer" &&
```
Comment (bharat619) [inline app/controllers/review_offer.js:31]: "Offer", this string check is made many times, probably u can add that in a constant. Not mandatory though

File: app/controllers/review_offer/items.js [hunk 2]
```diff
@@ -7,6 +7,29 @@ export default Ember.Controller.extend({
...
+        .get("lastObject");
+    }
+  ),
+
+  unreadOfferMessagesCount: Ember.computed(
+    "offer.messages",
+    "offer.messages.[]",
+    function() {
+      return this.get("offer.messages")
+        .sortBy("createdAt")
```
Comment (bharat619) [inline app/controllers/review_offer/items.js:26]: sortBy can be done after filtering

**Review by bharat619 — APPROVED** [review]
🚀

File: app/controllers/review_offer.js (hunk 1 above)
Comment (patrixr) [reply app/controllers/review_offer.js:31]: oh yeah, I'm rushing through

File: app/controllers/review_offer/items.js (hunk 2 above)
Comment (patrixr) [reply app/controllers/review_offer/items.js:26]: this one can actually be removed altogether, I'll do that

**Pull request merged**